### PR TITLE
Add regex support to --response-code

### PR DESCRIPTION
**Pull request by bootswithdefer**
#### General

- [X] Update Changelog following the conventions laid out [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [X] Update README with any necessary configuration snippets

- [X] Binstubs are created if needed

- [X] RuboCop passes

- [X] Existing tests pass

#### Purpose
Add regex support to --response-code in check-http
Add --response-code option to check-http-json

### CODE REVIEW

**Comment by majormoses** [comment]
Thanks for your contribution to Sensu plugins! Without people like you submitting PRs we couldn't run the project. I will review it shortly.

File: bin/check-http-json.rb [hunk 1]
```diff
@@ -73,6 +73,7 @@ class CheckJson < Sensu::Plugin::Check::CLI
   option :whole_response, short: '-w', long: '--whole-response', boolean: true, default: false
   option :dump_json, short: '-d', long: '--dump-json', boolean: true, default: false
   option :pretty, long: '--pretty', boolean: true, default: false
+  option :response_code, long: '--response-code CODE', default: '^2'
```
Comment (majormoses) [inline bin/check-http-json.rb]: please include a description ( I know not all the others do in this check but in general we try to make sure there is a description. In this case I would suggest including an indication that this accepts regex as opposed to a string. Also I am wondering after seeing #98 if we should while we are at it allow specifying multiple acceptable responses. That being said you could accomplish that with a slightly more complex regex such as `'^(200|429)'` while the other method of specifying multiple via comma separated is less powerful its also easier to use. I am open to both approaches so feel free to talk me into either approach, the one thing I would like to do is match it both locations so people are less likely to have issues if they use both.

Comment (bootswithdefer) [reply bin/check-http-json.rb]: The regex is more powerful and already used by other options like --query so the extra complexity shouldn't be a burden.

I will add it to check-http.rb and update the PR.

**Comment by bootswithdefer** [comment]
Description added, regex support added to check-http.rb

File: bin/check-http-json.rb [hunk 2]
```diff
@@ -74,6 +74,11 @@ class CheckJson < Sensu::Plugin::Check::CLI
   option :dump_json, short: '-d', long: '--dump-json', boolean: true, default: false
   option :pretty, long: '--pretty', boolean: true, default: false
 
+  option :response_code,
+         long: '--response-code REGEX',
+         description: 'Check for a specific response code REGEX',
+         default: '^2'
```
Comment (majormoses) [inline bin/check-http-json.rb]: I wonder if maybe `^2([0-9]{2})$` might be a better default regex as its more explicit.

File: bin/check-http-json.rb [hunk 3]
```diff
@@ -74,6 +74,11 @@ class CheckJson < Sensu::Plugin::Check::CLI
   option :dump_json, short: '-d', long: '--dump-json', boolean: true, default: false
   option :pretty, long: '--pretty', boolean: true, default: false
 
+  option :response_code,
+         long: '--response-code REGEX',
+         description: 'Check for a specific response code REGEX',
```
Comment (majormoses) [inline bin/check-http-json.rb]: This does not convey if the matching or unmatched HTTP status codes will have an OK status. Can we add something to clarify?

File: bin/check-http.rb [hunk 4]
```diff
@@ -205,8 +205,8 @@ class CheckHttp < Sensu::Plugin::Check::CLI
          proc: proc(&:to_i)
 
   option :response_code,
-         long: '--response-code CODE',
-         description: 'Check for a specific response code'
+         long: '--response-code REGEX',
+         description: 'Check for a specific response code REGEX'
```
Comment (majormoses) [inline bin/check-http.rb]: This does not convey if the matching or unmatched HTTP status codes will have an OK status. Can we add something to clarify?

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-http/versions/2.11.0

**Pull request merged**